gui/Graph.cpp

```diff
@@ -51,20 +43,34 @@ void Graph::draw() {
 			SDL_RenderDrawLine(this->renderer, currentX, currentY, nextX, nextY);
 		}
 		// TODO: if the objects are not functions, we have to draw a line between
-		// the last and the first point as well!
+		// the last and the first point as well: if (mode == "objects")
 	}
 
 	SDL_SetRenderDrawColor(this->renderer, 0, 0, 0, 255);
 }
 
+void Graph::addObject(Object *obj) {
+	debug("adding an object\n\n");
+	this->objects.push_back(obj);
+	int x = obj->getPoints()[0].x();
+	int y = obj->getPoints()[0].y();
```

Comment:
If you want the first point, you can use `obj->getPoint(1).x();` or even just `obj->getPoint().x();`

```
Point Object::getPoint () {
	return this->points[0];
}

Point Object::getPoint (int n) {
	if (0 < n-- && n < this->points.size()) return this->points[n];
	return Point();
}
```

Reply:
Great, I used the first option, because it is more explicit that way. But actually this is just for debugging purposes, just realized. Oh well, already commited it :woman_shrugging: